### ISSUE TITLE
Import lightning gives segmentation fault (core dumped)

### ISSUE THREAD

**Issue by abidikhairi** [Lightning-AI/lightning#17428]
## Issue Summary:
A segmentation fault (core dumped) error occurs when importing `lightning` module.

### File: test.py
```python

import torch
import lightning as L
from torch.nn import functional as F

# from project.models import mnist_classifier
# from project.datasets import MNISTDataModule

print("Hello World")
```

## Expected Behavior:
prints: "Hello World"

### What version are you seeing the problem on?

2.0+

### How to reproduce the bug

```python
1. install lightning: `pip install -U lightning`
2. run test.py
```

### Error messages and logs

```bash
zsh: segmentation fault (core dumped)  python project/test.py
```

### Environment

<details>
  <summary>Current environment</summary>

* CUDA:
	- GPU:
		- NVIDIA GeForce GTX 1650
	- available:         True
	- version:           11.6
* Lightning:
	- lightning:         2.0.1.post0
	- lightning-cloud:   0.5.33
	- lightning-utilities: 0.8.0
	- pytorch-lightning: 2.0.1.post0
	- torch:             1.12.1
	- torch-model-archiver: 0.6.0b20220513
	- torch-workflow-archiver: 0.2.4b20220513
	- torcharrow:        0.1.0
	- torchaudio:        0.12.1
	- torchmetrics:      0.9.3
	- torchserve:        0.6.0b20220513
	- torchtext:         0.13.1
	- torchvision:       0.13.1
* Packages:
	- absl-py:           1.2.0
	- accelerate:        0.15.0
	- agate:             1.6.3
	- agate-dbf:         0.2.2
	- agate-excel:       0.2.5
	- agate-sql:         0.5.8
	- aiohttp:           3.8.4
	- aiosignal:         1.2.0
	- alabaster:         0.7.13
	- alembic:           1.9.1
	- altair:            4.2.0
	- altair-data-server: 0.4.1
	- altair-saver:      0.5.0
	- altair-viewer:     0.4.0
	- aniso8601:         9.0.1
	- ansi2html:         1.8.0
	- antlr4-python3-runtime: 4.12.0
	- anyio:             3.6.1
	- appdirs:           1.4.4
	- argon2-cffi:       21.3.0
	- argon2-cffi-bindings: 21.2.0
	- arrow:             1.2.2
	- asttokens:         2.0.8
	- async-generator:   1.10
	- async-timeout:     4.0.2
	- attrs:             22.1.0
	- audioread:         3.0.0
	- autopage:          0.5.1
	- autopep8:          2.0.1
	- babel:             2.10.3
	- backcall:          0.2.0
	- backports.zoneinfo: 0.2.1
	- bcrypt:            4.0.1
	- beautifulsoup4:    4.11.1
	- bertviz:           1.4.0
	- biopython:         1.79
	- bleach:            5.0.1
	- blessed:           1.20.0
	- blinker:           1.5
	- blis:              0.7.9
	- boto3:             1.25.4
	- botocore:          1.28.4
	- bottle:            0.12.23
	- bravado:           11.0.3
	- bravado-core:      5.17.1
	- cachetools:        5.3.0
	- captum:            0.5.0
	- catalogue:         2.0.8
	- certifi:           2022.6.15
	- cffi:              1.15.1
	- chardet:           5.1.0
	- charset-normalizer: 2.1.1
	- click:             8.1.3
	- click-default-group: 1.2.2
	- cliff:             4.1.0
	- cloup:             0.13.1
	- cmaes:             0.9.0
	- cmd2:              2.4.2
	- cmdstanpy:         1.0.6
	- colorama:          0.4.6
	- colorlog:          6.7.0
	- colour:            0.1.5
	- commonmark:        0.9.1
	- confection:        0.0.3
	- configupdater:     3.1.1
	- contourpy:         1.0.5
	- convertdate:       2.4.0
	- coverage:          7.1.0
	- croniter:          1.3.14
	- cryptography:      38.0.3
	- csvkit:            1.0.7
	- cycler:            0.11.0
	- cymem:             2.0.7
	- cython:            0.29.32
	- dash:              2.7.1
	- dash-bootstrap-components: 0.13.1
	- dash-core-components: 2.0.0
	- dash-html-components: 2.0.0
	- dash-table:        5.0.0
	- dataclasses-json:  0.5.7
	- datasets:          2.7.1
	- dateutils:         0.6.12
	- dbfread:           2.0.7
	- de-core-news-sm:   3.4.0
	- debugpy:           1.6.3
	- decorator:         5.1.1
	- deepdiff:          6.3.0
	- defusedxml:        0.7.1
	- dill:              0.3.5.1
	- distlib:           0.3.6
	- docker-pycreds:    0.4.0
	- docopt:            0.6.2
	- docutils:          0.17.1
	- ecco:              0.1.2
	- einops:            0.4.1
	- en-core-web-sm:    3.4.1
	- entrypoints:       0.4
	- ephem:             4.1.3
	- et-xmlfile:        1.1.0
	- evaluate:          0.3.0
	- exceptiongroup:    1.1.0
	- executing:         0.10.0
	- falcon:            2.0.0
	- fastapi:           0.88.0
	- fastjsonschema:    2.16.1
	- ffmpeg-python:     0.2.0
	- ffmpy:             0.3.0
	- filelock:          3.9.0
	- findspark:         2.0.1
	- fire:              0.5.0
	- flake8:            6.0.0
	- flask:             2.2.2
	- flask-restful:     0.3.9
	- fonttools:         4.36.0
	- fqdn:              1.5.1
	- fr-core-news-sm:   3.4.0
	- frozenlist:        1.3.1
	- fsspec:            2022.7.1
	- future:            0.18.2
	- fuzzywuzzy:        0.18.0
	- gensim:            4.2.0
	- gitdb:             4.0.9
	- gitpython:         3.1.27
	- glcontext:         2.3.6
	- google-auth:       2.11.0
	- google-auth-oauthlib: 0.4.6
	- gputil:            1.4.0
	- gradio:            3.11.0
	- greenlet:          2.0.1
	- grpcio:            1.47.0
	- h11:               0.12.0
	- hijri-converter:   2.2.4
	- holidays:          0.15
	- htbuilder:         0.6.1
	- httpcore:          0.15.0
	- httpx:             0.23.1
	- hug:               2.6.1
	- huggingface-hub:   0.11.0
	- idna:              3.3
	- imagesize:         1.4.1
	- importlib-metadata: 4.12.0
	- importlib-resources: 5.9.0
	- iniconfig:         1.1.1
	- inquirer:          3.1.3
	- ipcmagic-cscs:     1.1.0
	- ipykernel:         6.15.1
	- ipyparallel:       8.4.1
	- ipython:           8.4.0
	- ipython-genutils:  0.2.0
	- isodate:           0.6.1
	- isoduration:       20.11.0
	- isosurfaces:       0.1.0
	- itsdangerous:      2.1.2
	- jedi:              0.18.1
	- jinja2:            3.1.2
	- jiwer:             2.5.1
	- jmespath:          1.0.1
	- joblib:            1.1.0
	- json5:             0.9.10
	- jsonpointer:       2.3
	- jsonref:           1.0.0.post1
	- jsonschema:        4.14.0
	- jupyter-client:    7.3.4
	- jupyter-core:      4.11.1
	- jupyter-server:    1.18.1
	- jupyterlab:        3.4.5
	- jupyterlab-pygments: 0.2.2
	- jupyterlab-server: 2.15.0
	- kafka-python:      2.0.2
	- kaggle:            1.5.12
	- kiwisolver:        1.4.4
	- korean-lunar-calendar: 0.2.1
	- langchain:         0.0.129
	- langcodes:         3.3.0
	- leather:           0.3.4
	- levenshtein:       0.20.2
	- librosa:           0.9.2
	- lightning:         2.0.1.post0
	- lightning-cloud:   0.5.33
	- lightning-utilities: 0.8.0
	- linkify-it-py:     1.0.3
	- llvmlite:          0.39.1
	- lunarcalendar:     0.0.9
	- lxml:              4.9.1
	- mako:              1.2.4
	- manim:             0.16.0.post0
	- manim-ml:          0.0.20
	- manimpango:        0.4.1
	- mapbox-earcut:     0.12.11
	- markdown:          3.4.1
	- markdown-it-py:    2.1.0
	- markupsafe:        2.1.1
	- marshmallow:       3.19.0
	- marshmallow-enum:  1.5.1
	- matplotlib:        3.6.2
	- matplotlib-inline: 0.1.6
	- mccabe:            0.7.0
	- mdit-py-plugins:   0.3.1
	- mdurl:             0.1.2
	- memory-profiler:   0.61.0
	- metakernel:        0.29.2
	- metis:             0.2a5
	- mistune:           0.8.4
	- mkl-fft:           1.3.1
	- mkl-random:        1.2.2
	- mkl-service:       2.4.0
	- moderngl:          5.6.4
	- moderngl-window:   2.4.1
	- monotonic:         1.6
	- more-itertools:    9.0.0
	- msgpack:           1.0.4
	- multidict:         6.0.2
	- multipledispatch:  0.6.0
	- multiprocess:      0.70.13
	- munkres:           1.1.4
	- murmurhash:        1.0.9
	- mypy-extensions:   0.4.3
	- nbclassic:         0.4.3
	- nbclient:          0.6.6
	- nbconvert:         6.5.3
	- nbformat:          5.4.0
	- nest-asyncio:      1.5.5
	- networkx:          2.8.5
	- nltk:              3.7
	- node2vec:          0.4.6
	- notebook:          6.4.12
	- notebook-shim:     0.1.0
	- numba:             0.56.2
	- numpy:             1.23.1
	- nvgpu:             0.9.0
	- oauthlib:          3.2.0
	- olefile:           0.46
	- openai:            0.27.2
	- openpyxl:          3.0.10
	- optuna:            2.10.1
	- ordered-set:       4.1.0
	- orjson:            3.8.2
	- outcome:           1.2.0
	- packaging:         23.0
	- pandas:            1.3.5
	- pandocfilters:     1.5.0
	- paramiko:          2.12.0
	- parsedatetime:     2.4
	- parso:             0.8.3
	- pathtools:         0.1.2
	- pathy:             0.6.2
	- patsy:             0.5.3
	- pbr:               5.11.0
	- pexpect:           4.8.0
	- pickleshare:       0.7.5
	- pillow:            9.2.0
	- pip:               22.1.2
	- pkgutil-resolve-name: 1.3.10
	- platformdirs:      2.6.2
	- plotly:            5.10.0
	- pluggy:            1.0.0
	- pooch:             1.6.0
	- portpicker:        1.5.2
	- preshed:           3.0.8
	- prettytable:       3.5.0
	- prometheus-client: 0.14.1
	- promise:           2.3
	- prompt-toolkit:    3.0.30
	- prophet:           1.1
	- protobuf:          3.19.4
	- psutil:            5.9.1
	- psycopg2-binary:   2.9.6
	- ptyprocess:        0.7.0
	- pudb:              2022.1.2
	- pure-eval:         0.2.2
	- pwgen:             0.8.2.post0
	- py:                1.11.0
	- py-cpuinfo:        9.0.0
	- pyarrow:           9.0.0
	- pyasn1:            0.4.8
	- pyasn1-modules:    0.2.8
	- pycairo:           1.21.0
	- pycodestyle:       2.10.0
	- pycparser:         2.21
	- pycryptodome:      3.16.0
	- pydantic:          1.10.2
	- pydata-sphinx-theme: 0.8.1
	- pydeck:            0.8.0b3
	- pydeprecate:       0.3.2
	- pydub:             0.25.1
	- pyee:              9.0.4
	- pyflakes:          3.0.1
	- pyglet:            1.5.26
	- pygments:          2.13.0
	- pyjwt:             2.6.0
	- pymeeus:           0.5.11
	- pympler:           1.0.1
	- pynacl:            1.5.0
	- pynvml:            11.4.1
	- pyparsing:         3.0.9
	- pyperclip:         1.8.2
	- pyproject-api:     1.5.0
	- pyrr:              0.10.3
	- pyrsistent:        0.18.1
	- pyscaffold:        4.4
	- pysocks:           1.7.1
	- pytest:            7.1.2
	- python-dateutil:   2.8.2
	- python-editor:     1.0.4
	- python-ffmpeg:     1.0.15
	- python-multipart:  0.0.5
	- python-slugify:    6.1.2
	- pytimeparse:       1.1.8
	- pytorch-lightning: 2.0.1.post0
	- pytz:              2022.2.1
	- pytz-deprecation-shim: 0.1.0.post0
	- pyyaml:            5.4.1
	- pyzmq:             23.2.1
	- rapidfuzz:         2.13.2
	- readchar:          4.0.5
	- redis:             4.5.4
	- regex:             2022.8.17
	- reportlab:         3.6.12
	- requests:          2.28.1
	- requests-oauthlib: 1.3.1
	- resampy:           0.4.1
	- responses:         0.18.0
	- rfc3339-validator: 0.1.4
	- rfc3986:           1.5.0
	- rfc3987:           1.3.8
	- rich:              12.5.1
	- rsa:               4.9
	- rst2pdf:           0.99
	- s3transfer:        0.6.0
	- scikit-learn:      0.24.2
	- scipy:             1.8.1
	- screeninfo:        0.8
	- seaborn:           0.12.1
	- selenium:          4.7.2
	- semver:            2.13.0
	- send2trash:        1.8.0
	- sentencepiece:     0.1.97
	- sentry-sdk:        1.9.5
	- setproctitle:      1.3.2
	- setuptools:        65.6.3
	- setuptools-git:    1.2
	- setuptools-scm:    7.1.0
	- shortuuid:         1.0.9
	- simplejson:        3.17.6
	- six:               1.16.0
	- skia-pathops:      0.7.2
	- smart-open:        5.2.1
	- smartypants:       2.0.1
	- smmap:             5.0.0
	- sniffio:           1.2.0
	- snowballstemmer:   2.2.0
	- sortedcontainers:  2.4.0
	- soundfile:         0.10.3.post1
	- soupsieve:         2.3.2.post1
	- spacy:             3.4.2
	- spacy-legacy:      3.0.10
	- spacy-loggers:     1.0.3
	- sphinx:            4.5.0
	- sphinx-book-theme: 0.3.3
	- sphinx-rtd-theme:  1.2.0
	- sphinxcontrib-applehelp: 1.0.4
	- sphinxcontrib-devhelp: 1.0.2
	- sphinxcontrib-htmlhelp: 2.0.1
	- sphinxcontrib-jquery: 2.0.0
	- sphinxcontrib-jsmath: 1.0.1
	- sphinxcontrib-mermaid: 0.8
	- sphinxcontrib-qthelp: 1.0.3
	- sphinxcontrib-serializinghtml: 1.1.5
	- spylon:            0.3.0
	- spylon-kernel:     0.4.1
	- sqlalchemy:        1.4.44
	- srsly:             2.4.5
	- srt:               3.5.2
	- st-annotated-text: 3.0.0
	- stack-data:        0.4.0
	- starlette:         0.22.0
	- starsessions:      1.3.0
	- statsmodels:       0.13.5
	- stevedore:         4.1.1
	- streamlit:         1.13.0
	- swagger-spec-validator: 3.0.2
	- tabulate:          0.8.10
	- tenacity:          8.1.0
	- tensorboard:       2.10.0
	- tensorboard-data-server: 0.6.1
	- tensorboard-plugin-wit: 1.8.1
	- termcolor:         1.1.0
	- terminado:         0.15.0
	- text-unidecode:    1.3
	- thinc:             8.1.5
	- threadpoolctl:     3.1.0
	- tinycss2:          1.1.1
	- tokenizers:        0.12.1
	- toml:              0.10.2
	- tomli:             2.0.1
	- tomlkit:           0.11.6
	- toolz:             0.12.0
	- torch:             1.12.1
	- torch-model-archiver: 0.6.0b20220513
	- torch-workflow-archiver: 0.2.4b20220513
	- torcharrow:        0.1.0
	- torchaudio:        0.12.1
	- torchmetrics:      0.9.3
	- torchserve:        0.6.0b20220513
	- torchtext:         0.13.1
	- torchvision:       0.13.1
	- tornado:           6.2
	- tox:               4.4.4
	- tqdm:              4.64.0
	- traitlets:         5.3.0
	- transformers:      4.24.0
	- transformix:       0.0.post1.dev11+g04f1134.d20230208
	- trio:              0.22.0
	- trio-websocket:    0.9.2
	- ttygif:            1.2.640
	- twilio:            7.16.2
	- typer:             0.4.2
	- typing:            3.7.4.3
	- typing-extensions: 4.3.0
	- typing-inspect:    0.8.0
	- tzdata:            2022.4
	- tzlocal:           4.2
	- uc-micro-py:       1.0.1
	- ujson:             5.4.0
	- uri-template:      1.2.0
	- urllib3:           1.26.13
	- urwid:             2.1.2
	- urwid-readline:    0.13
	- uvicorn:           0.20.0
	- validators:        0.20.0
	- virtualenv:        20.17.1
	- waitress:          2.1.2
	- wandb:             0.13.2
	- wasabi:            0.10.1
	- watchdog:          2.1.9
	- wcwidth:           0.2.5
	- webcolors:         1.12
	- webencodings:      0.5.1
	- websocket-client:  1.3.3
	- websockets:        10.4
	- werkzeug:          2.2.2
	- wheel:             0.37.1
	- whisper:           1.0
	- wsproto:           1.2.0
	- xlrd:              2.0.1
	- xxhash:            3.0.0
	- yapf:              0.32.0
	- yarl:              1.8.1
	- zipp:              3.8.1
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- ELF
	- processor:         
	- python:            3.8.13
	- version:           #1 SMP PREEMPT Thu Mar 23 01:40:42 UTC 2023

</details>

**Comment by awaelchli** [comment]
@abidikhairi Have you verified that it is the import of lightning and not the torch import that is causing the segfault? Is there indication of which import inside lightning is causing this?

**Comment by abidikhairi** [comment]
Hi @awaelchli, 

Yes it is the import of `lightning` that causes the seffault. Can you point out how to investigate if submodules are casusing this segfault error.

However, i've create a new python env with `torch=1.13.1`, then installed lightning and worked like a charm. Consequently, i assume that my env is broken. i will investigate and update this thread.

Thank you for your time.
